core/Tracker.php

```diff
@@ -361,4 +377,43 @@ private static function isDebugEnabled()
 
         return false;
     }
-}
+
+    private function outputTrackerConfigs()
+    {
+        $configs = [];
+
+        /**
+         * Triggered when returning tracker configuration to the JavaScript tracker. Some plugins' tracking code may
+         * depend on information that is only stored server side. Use this event to provide this information to the
+         * JavaScript tracker.
+         *
+         * Since this event is invoked during tracking, performance is very important. Use the tracker cache
+         * as much as possible and avoid making direct requests to the database if it can be avoided.
+         *
+         * @param array &$configs Array mapping plugin names with their respective configuration. For example:
+         *                        ```
+         *                        [
+         *                            'MyPlugin' => [ 'setting1' => 1, 'setting2' => 3 ],
+         *                            'MyOtherPlugin' => ...,
+         *                        ]
+         *                        ```
+         */
+        Piwik::postEvent('Tracker.getTrackerConfigs', [&$configs]);
```

Comment:
not too important, but might be good to post some additional params here like we have in some places. eg `idSite`? eg `Piwik::postEvent('Tracker.getTrackerConfigs', [&$configs, array('idSite' => $idSite)]);` or `Piwik::postEvent('Tracker.getTrackerConfigs', [&$configs, $idSite]);`? This way don't have to get the idSite in every event and might be easier to test it.

Reply:
Done.